### PR TITLE
Add authentication with jsonwebtoken and cookies

### LINKED ISSUE #10: Access tokens
As a user,
I like to control access to logs,
so that there is no unauthorized access.

- jwt

### PR DESCRIPTION
**Pull request by lmachens**
Resolve #10 

There is no password required for authentication right now. I plan to allow access by email only and add email verification in the future.
New routes:
`/api/register` will create a new account based on an email.
`/api/login` will login by email and sends a cookie to the client (jsonwebtoken) to verify the access in upcoming requests.
`/api/logs` GET checks for a valid jsonwebtoken in the cookies before allowing access to the logs.
The POST route is not effected right now.

UI is very basic. Next, I'll focus on the UI.

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -1,10 +1,20 @@
 import React from "react";
 import GlobalStyles from "./GlobalStyles";
-import { BrowserRouter as Router, Switch, Route, Link } from "react-router-dom";
+import {
+  BrowserRouter as Router,
```
Comment (macfrei) [inline client/src/App.js:4]: What is your reason to have your `Router` in the `App.js` instead of `index.js`?

File: client/src/App.js [hunk 2]
```diff
@@ -1,10 +1,20 @@
...
 import Apps from "./pages/Apps";
 import Logs from "./pages/Logs";
+import useCookie from "./hooks/useCookie";
+import Login from "./pages/Login";
 
 function App() {
+  const [authToken] = useCookie("authToken");
+
   return (
     <div>
```
Comment (macfrei) [inline client/src/App.js:19]: Is there a reason to use a `div`-element instead of fragments?

**Review by macfrei — COMMENTED** [review]
I just have some small questions/comments, other than that it looks good to me. 👍 ;-)

File: client/src/App.js (hunk 2 above)
Comment (lmachens) [reply client/src/App.js:19]: Good catch. Not really, I will remove it soon.

File: client/src/App.js (hunk 1 above)
Comment (lmachens) [reply client/src/App.js:4]: In my understanding, the `Router` is part of my app. The app wouldn't work without the router.
`index.js` is the main entry point and says which app should be rendered in which container. In addition, this is a good place for other services which are not related to the React app (like the serviceWorker for Progressive Web Apps). 

You could render apps multiple times:

```jsx
ReactDOM.render(
  <App />,
  document.getElementById("first")
);

ReactDOM.render(
  <App />,
  document.getElementById("second")
);

```

Comment (macfrei) [reply client/src/App.js:4]: That does make sense!

**Pull request merged**